### DISCUSSION TITLE
How to disable some keybindings (delete buffers)

### DISCUSSION THREAD

**Discussion by morenohernan** [LazyVim/LazyVim#1407] [Q&A] [answered]
I'm a beginner to NeoVim. I want to disable some keybindings but it seems it isn't working. 
<img width="782" alt="image" src="https://github.com/LazyVim/LazyVim/assets/700619/054959d9-551d-449e-98f2-e236be8d359c">

In my keymaps.lua .. I have this:

-- To delete buffers
vim.keymap.del("n", "<leader>bD") **-- I want to delete this keybind but still I can see it**
vim.keymap.del("n", "<leader>bP") **-- I want to delete this keybind but still I can see it**

map("n", "<leader>bd",'delete-buffers', { desc = "Delete buffers"}) **-- I want to override with this desc but it seems it is not taking it**
map("n", "<leader>bdr", "<cmd>BufferLineCloseRight<cr>", { desc = "Delete buffers to the right from current" })
map("n", "<leader>bdl", "<cmd>BufferLineCloseLeft<cr>", { desc = "Delete buffers to the left from current" })
map("n", "<leader>bdo", "<cmd>BufferLineCloseOthers<cr>", { desc = "Delete buffers others than current" })
map("n", "<leader>bdc", function() require("mini.bufremove").delete(0, false) end, { desc = "Delete current buffer" })
map("n", "<leader>bdC", function() require("mini.bufremove").delete(0, true) end, { desc = "Delete current buffer (Force)" })

Any hint?

**Comment by dpetka2001** [comment]
The keybindings are `<leader>bD`, not just `bD`. If you change that you should be able to delete the keybindings you want. For the description, you could do in a file in `/lua/plugins` directory something like 
```lua
return {
      {
        "folke/which-key.nvim",
        opts = {
          defaults = {
            ["<leader>bd"] = { name = "+delete-buffers" },
          },
        },
      },
}
```

**Comment by morenohernan** [comment]
Thanks for the hint, @dpetka2001 . So far it has changed the description as you mentioned. However, not removed the  `<leader>bP`

```
return {
      {
        "folke/which-key.nvim",
        opts = {
          defaults = {
            ["<leader>bd"] = { name = "+delete-buffers" },
            ["<leader>bs"] = { name = "+sort-buffers" },
          },
        },
      },
}
```

keymaps.lua
------------
<img width="1374" alt="image" src="https://github.com/LazyVim/LazyVim/assets/700619/b78061b1-14b6-4bae-b96a-6afc54bd191f">


In Lazyvim
-----------
<img width="622" alt="image" src="https://github.com/LazyVim/LazyVim/assets/700619/3afc5790-707f-4afd-a43d-b6b94cde5ae0">

**Reply by dpetka2001** [reply]
In the `which-key` spec you only define the description that you want. You should still delete/change the keybindings you want with `vim.keymap.del/set` respectively.

**Reply by dpetka2001** [reply, marked as answer]
One correction. `<leader>bd` is configured in https://github.com/LazyVim/LazyVim/blob/f9dadc11b39fb0b027473caaab2200b35c9f0c8b/lua/lazyvim/plugins/editor.lua#L401 
and `<leader>bP` in https://github.com/LazyVim/LazyVim/blob/f9dadc11b39fb0b027473caaab2200b35c9f0c8b/lua/lazyvim/plugins/ui.lua#L58-L59.
So, you would effectively disable the keybindings using the `keys` section in the corresponding specs. See the [docs](https://www.lazyvim.org/configuration/plugins#%EF%B8%8F-adding--disabling-plugin-keymaps) for how to do that and then change them to do whatever you want.

**Reply by morenohernan** [reply]
Thanks a lot for the help, @dpetka2001 . It is working now.
